api/graphite.go

```diff
@@ -107,11 +107,17 @@ func (s *Server) findSeries(ctx context.Context, orgId uint32, patterns []string
 
 		// assign a fractional maxSeries limit (not global, but relative to how much data the peer has)
 		// look at each shardgroup and check how many partitions it has
-		// (we assume each shardgroup is consistent across different peers for that shardgroup)
+		// we have 2 assumptions here
+		// 1) on the cluster topology: each shardgroup is consistent across different peers for that shardgroup.
+		//    (e.g. you can't have one node with shards [2,3] and another one with just [2] or with [3,4], it should be [2,3] each.
+		// 2) any dataset is evenly distributed across shards. e.g. if foo.* matches 1000 series and you have 4 shardgroups, then we assume
+		//    that each shardgroup holds exactly 250 series. In practice we are fairly close, but rarely exact. In some unlikely scenarios
+		//    this may mean unfairly being prematurely rejected.
```

Comment:
Is it worth adding in some small percentage of tolerance per instance (e.g. 5%)? This would still allow a request to be aborted at a statistically likely breached limit, but would reduce the chance of unfair rejection do to (unlucky) data distribution.

Reply:
Per shard group limit is probably something I will find useful to add to the tagged version as well.

Reply:
> Is it worth adding in some small percentage of tolerance per instance (e.g. 5%)? This would still allow a request to be aborted at a statistically likely breached limit, but would reduce the chance of unfair rejection do to (unlucky) data distribution.

I contemplated the same thing, but I ultimately think of the `max-series-per-req` setting as a fairly heavy-handed/imprecise measure, where operators should not just set it a tad higher than what legitimate requests routinely need, but significantly higher. to account for not just this, but all other kinds of trade-offs and caveats.